### ISSUE TITLE
Screenshot library creates Segmentation Fault when using with WXPython 4.0.7 in Linux Mint

### ISSUE THREAD

**Issue by HelioGuilherme66** [robotframework/robotframework#3403]
On a Virtual Box with Linux Mint Debian edition (and also reported on Ubuntu 18.04), Python 2.7.13, wxPython 4.0.7.post2.
Using Screenshot library works fine with PyGTK but causes segmentation fault when using wxPython.
```
$ python -m robot.libraries.Screenshot example.jpg pygtk
Using 'gtk' to take screenshot.
Taking test screenshot to 'example.jpg'.
Success!
$ python -m robot.libraries.Screenshot example.jpg wxpython
Using 'wx' to take screenshot.
Taking test screenshot to 'example.jpg'.
Segmentation fault
$
```

This problem have the side effect of making RIDE blocked, even if the import is using other than wxPython screenshot_module. [RIDE #2070](https://github.com/robotframework/RIDE/issues/2070)

**Comment by pekkaklarck** [comment]
Can you investigate where the problem occurs? Has wxPython API changed?

**Comment by HelioGuilherme66** [comment]
I will investigate.

**Comment by HelioGuilherme66** [comment]
@pekkaklarck Seems that when using Screenshot library with wxPython and Python 2.7 on Linux, users must install the GTK2 version of wxPython, including for the fix on my PR.
I have posted on wxPython group [this](https://discuss.wxpython.org/t/linux-wheels-for-gtk3-are-calling-gtk2-libraries-on-python-2-7/34293/2).

**Comment by pekkaklarck** [comment]
Thanks for debugging @HelioGuilherme66! Luckily the needed changes were really small. Some comments/questions below:

1. Could you take a look at my comment to the PR about only creating the app reference when needed?

2. Does this only occur with Python 2.7, not with Python 3?

3. If certain wxPython versions don't work, that should be mentioned in the library documentation when talking about using wxPython in general. Could you add such a note if needed?

**Comment by HelioGuilherme66** [comment]
1. I have tried creating the App in the _wx_screenshot function, and it did not solve the problem.
2. This only occurs with Python 2.7.
3. If the PR is merged, then no need for documentation notes.

**Comment by pekkaklarck** [comment]
As I commented the PR, I'm still somewhat worried about the global `wx.App` reference but I guess that's not a big issue because it's only created if `wx` can be imported in the first place.

**Comment by pekkaklarck** [comment]
The fix for this issue caused #3798 and the fix was reverted in RF 4.1. It's unlikely there are many users who use wxPython on Python 2 anymore (I couldn't even install it there), so even if reverting the fix would introduce this bug again it shouldn't affect too many users.
